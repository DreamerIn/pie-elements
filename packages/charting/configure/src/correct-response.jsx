import * as React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Checkbox from '@material-ui/core/Checkbox';
import { Chart } from '@pie-lib/charting';


import { set } from 'lodash';
import CheckBox from '@material-ui/core/es/internal/svg-icons/CheckBox';

const styles = theme => ({
  container: {
    border: '2px solid #ababab',
    borderRadius: '4px',
    padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 4}px`,
    background: '#fafafa'
  },
  button: {
    marginTop: theme.spacing.unit * 3,
    cursor: 'pointer',
    background: '#eee',
    padding: theme.spacing.unit * 2,
    width: 'fit-content',
    borderRadius: '4px'
  }
});

export class CorrectResponse extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    model: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired
  };

  changeData = (mark, data) => {
    const { answers } = this.props.model || {};

    answers[mark].data = data;
    this.props.onChange({
      ...this.props.model,
      answers
    });
  };

  changeEditable = (mark, value) => {
    const { answers } = this.props.model || {};

    answers[mark].editCategoryEnabled = value;
    this.props.onChange({
      ...this.props.model,
      answers
    });
  };

  changeAddRemoveEnabled = (mark, value) => {
    const { answers } = this.props.model || {};

    answers[mark].addCategoryEnabled = value;
    this.props.onChange({
      ...this.props.model,
      answers
    });
  };


  render() {
    const { classes, model, onChange, charts } = this.props;

    return (
      <div>
        Define Correct Response
        <div className={classes.container}>
          {Object.keys(model.answers).map(mark => (
            <div key={`correct-response-graph-${model.answers[mark].name}`}>
              <p>{model.answers[mark].name}</p>
              <Chart
                chartType={model.chartType}
                size={model.graph}
                domain={model.domain}
                range={model.range}
                charts={charts}
                data={model.answers[mark].data}
                title={model.title}
                onDataChange={(data) => this.changeData(mark, data)}
                editCategoryEnabled={model.answers[mark].editCategoryEnabled}
                addCategoryEnabled={model.answers[mark].addCategoryEnabled}
                categoryDefaultLabel={model.categoryDefaultLabel}
              />
              <div>
                <Checkbox
                  checked={model.answers[mark].editCategoryEnabled}
                  onChange={(e) => {
                    this.changeEditable(mark, e.target.checked)
                  }}
                />
                Student can edit category labels
              </div>
              <div>
                <Checkbox
                  checked={model.answers[mark].addCategoryEnabled}
                  onChange={(e) => {
                    this.changeAddRemoveEnabled(mark, e.target.checked)
                  }}
                />
                Student can add/remove category labels
              </div>
            </div>
          ))}

          <div
            className={classes.button}
            onClick={() => {
              set(
                model,
                `answers.${`alternate${Object.keys(model.answers).length}`}`,
                {
                  name: `Alternate ${Object.keys(model.answers).length}`,
                  data: [],
                  editCategoryEnabled: true,
                  addCategoryEnabled: true
                }
              );
              onChange(model);
            }}
          >
            ADD ALTERNATE
          </div>
        </div>
      </div>
    );
  }
}

export default withStyles(styles)(CorrectResponse);
