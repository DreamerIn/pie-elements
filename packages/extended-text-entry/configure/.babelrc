{
  "sourceMaps": "inline",
  "presets": [
    "env","react","stage-0","airbnb"
  ],
  "plugins": [
    "transform-class-properties"
  ]  
}