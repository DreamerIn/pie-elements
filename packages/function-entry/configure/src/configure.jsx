import { withStyles } from 'material-ui/styles';
import { FeedbackConfig, Checkbox } from '@pie-lib/config-ui';
import { HintsPopover } from '@pie-ui/function-entry';
import React from 'react';
import PropTypes from 'prop-types';
import debug from 'debug';
import Input from 'material-ui/Input';
import Typography from 'material-ui/Typography';
import ModelConfig from './model-config';
import { modelToFeedbackConfig, feedbackConfigToModel } from './feedback-mapper';

const log = debug('@pie-element:text-entry:configure');

const styles = theme => ({
  title: {
    'font-size': '1.1rem',
    display: 'block',
    'margin-top': theme.spacing.unit * 2,
    'margin-bottom': theme.spacing.unit,
  },
  'equation-row-container': {
    display: 'flex',
    'align-items': 'center',
    'justify-content': 'center',
    'margin-top': theme.spacing.unit * 2,
    'margin-bottom': theme.spacing.unit * 2,
  },
  'equation-label': {
    'margin-right': theme.spacing.unit,
  },
});

class Configure extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    model: PropTypes.any,
    onModelChanged: PropTypes.func,
  };

  constructor(props) {
    super(props);

    this.state = {
      hintsAnchorEl: null,
      anchorReference: 'anchorEl',
      hintsOpen: true,
    };
  }

  onModelConfigChange = (cfg) => {
    this.props.model.model = cfg;
    this.props.onModelChanged(this.props.model);
  }

  onEquationChange = (event) => {
    this.props.model.correctResponse.equation = event.target.value;
    this.props.onModelChanged(this.props.model);
  }

  onFormattingHelpChange = (event) => {
    this.props.model.showFormattingHelp = event.target.value;
    this.props.onModelChanged(this.props.model);
  }

  handleHintsPopoverOpen = event => {
    this.setState({ hintsAnchorEl: event.target });
  };

  handleHintsPopoverClose = () => {
    this.setState({ hintsAnchorEl: null });
  };

  onFeedbackChange = (feedbackConfig) => {
    const model = feedbackConfigToModel(feedbackConfig, this.props.model);
    this.props.onModelChanged(this.props.model);
  }

  render() {
    const { classes, model } = this.props;
    const { anchorReference, hintsAnchorEl } = this.state;
    const feedbackConfig = modelToFeedbackConfig(model);
    const hintsOpen = !!hintsAnchorEl;

    log('[render] model', model);

    debugger;

    return (
        <div>
          <Typography type="body1">
            <span>In Evaluate an Expression, a student submits a linear or polynomial expression to be evaluated.</span>
            <span className={classes.title}>Expression</span>
            <span>Enter the expression against which the the student&#39;s response will be evaluated. <br/>
              Note that <b><i> y </i></b>is the dependent variable and <b><i> f(x) </i></b> is some function
              where <b><i> x </i></b> is the independent variable. <br/>
              Expressions <b> must </b> be input using <b><i> x </i></b> and/or <b><i> y </i></b> variables.
            </span>
          </Typography>
          <div className={classes['equation-row-container']}>
            <Typography type="body1">
              <span className={classes['equation-label']}>y = </span>
            </Typography>
            <Input
              type="text"
              className="equation-input"
              onChange={this.onEquationChange}
              value={model.correctResponse.equation}
              placeholder="Enter the expression here."
            />
          </div>
          <Typography type="body1">
            <span>
              When the student submits an answer the answer will be evaluated against the expression by
              generating test points. The test points are created by replacing the <i>x</i> value within the
              function with random whole numbers within the domain. The <i>y</i> value is then determined by
              evaluating the equation using the javascript eval function. This is done many times (~50) in order
              to be sure of the correctness.
            </span>
          </Typography>
          <div className={classes['hints-control-row']}>
            <Checkbox
              className={classes['hints-checkbox']}
              checked={model.model.showFormattingHelp}
              onChange={this.onFormattingHelpChange}
              label=""
            />
            <Typography>
              <span>
                Show the student the formatting &nbsp;
                <span
                    onMouseOver={this.handleHintsPopoverOpen}
                    onMouseOut={this.handleHintsPopoverClose}
                >
                  <i><b>hints</b></i>
                </span>
                &nbsp; for constructing an answer
              </span>
            </Typography>
          </div>
          <HintsPopover
            hintsOpen={hintsOpen}
            hintsAnchorEl={hintsAnchorEl}
            anchorReference={anchorReference}
          />
          <ModelConfig config={model.model} onChange={this.onModelConfigChange} />
          <FeedbackConfig
              feedback={feedbackConfig}
              onChange={this.onFeedbackChange} />
        </div>
    )
  }
}

const ConfigureMain = withStyles(styles)(Configure);

class StateWrapper extends React.Component {
  static propTypes = {
    model: PropTypes.any,
    onModelChanged: PropTypes.func,
  };

  constructor(props) {
    super(props);

    this.state = {
      model: props.model
    }

    this.onModelChanged = (m) => {
      this.setState({ model: m }, () => {
        this.props.onModelChanged(this.state.model);
      });
    }
  }

  render() {
    const { model } = this.state;
    return <ConfigureMain model={model} onModelChanged={this.onModelChanged}/>
  }
}

export default StateWrapper;
