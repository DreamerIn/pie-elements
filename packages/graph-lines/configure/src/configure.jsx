import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import ChartConfig from '@pie-lib/charting-config';
import { FeedbackConfig } from '@pie-lib/config-ui';
import PropTypes from 'prop-types';
import debug from 'debug';
import Typography from '@material-ui/core/Typography';
import GeneralConfigBlock from './general-config-block';

const log = debug('@pie-element:graph-lines:configure');

const styles = theme => ({
  title: {
    fontSize: '1.1rem',
    display: 'block',
    marginTop: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit
  },
  content: {
    marginTop: theme.spacing.unit * 2
  }
});

class Configure extends React.Component {
  static propTypes = {
    onModelChanged: PropTypes.func,
    classes: PropTypes.object,
    model: PropTypes.object.isRequired
  };

  constructor(props) {
    super(props);

    this.defaults = JSON.parse(JSON.stringify(props.model));
  }

  resetToDefaults = () => {
    this.props.onModelChanged(JSON.parse(JSON.stringify(this.defaults)));
  };

  onChange = (config, name) => {
    this.props.model.model.config = { ...config };

    // if the correct line changed, we need to update the correct response too on the model
    if (name === 'correctLine') {
      this.props.model.correctResponse = config.correctLine;
    }

    this.props.onModelChanged(this.props.model);
  };

  onFeedbackChange = feedback => {
    const { model, onModelChanged } = this.props;
    model.feedback = feedback;
    onModelChanged(model);
  };

  render() {
    const { classes, model } = this.props;
    const config = model.model.config;

    log('[render] model', model);

    return (
      <div className={classes.content}>
        <Typography component="div" type="body1">
          <span>
            This interaction asks a student to draw a line that meets specific criteria.
            The student will draw the line by clicking on two points on the graph.
          </span>
          <h2>Lines</h2>
          <span>Line equations must be in y=mx+b form. Only whole number coordinates can be plotted.</span>
        </Typography>
        <GeneralConfigBlock
          config={config}
          onChange={this.onChange}
        />
        <ChartConfig
          config={config}
          onChange={this.onChange}
          resetToDefaults={this.resetToDefaults}
        />
        <FeedbackConfig
          allowPartial={false}
          feedback={model.feedback}
          onChange={this.onFeedbackChange}
        />
      </div>
    );
  }
}

const ConfigureMain = withStyles(styles)(Configure);

class StateWrapper extends React.Component {
  static propTypes = {
    model: PropTypes.any,
    onModelChanged: PropTypes.func
  };

  constructor(props) {
    super(props);

    this.state = {
      model: props.model
    };

    this.onModelChanged = m => {
      this.setState({ model: m }, () => {
        this.props.onModelChanged(this.state.model);
      });
    };
  }

  render() {
    const { model } = this.state;
    return <ConfigureMain model={model} onModelChanged={this.onModelChanged} />;
  }
}

export default StateWrapper;
