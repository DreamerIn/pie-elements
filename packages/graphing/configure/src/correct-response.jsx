import * as React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { GraphContainer as Graph, tools } from '@pie-lib/graphing';

import { get, set } from 'lodash';

const styles = theme => ({
  column: {
    flex: 1
  },
  graphingTools: {
    marginTop: theme.spacing.unit * 3,
    color: '#ababab'
  },
  availableTools: {
    marginTop: theme.spacing.unit * 3,
    marginBottom: theme.spacing.unit * 3,
    display: 'flex'
  },
  availableTool: {
    cursor: 'pointer',
    margin: theme.spacing.unit,
    padding: theme.spacing.unit,
    textTransform: 'capitalize'
  },
  selectedTool: {
    background: '#d8d8d8',
    border: '2px solid #ababab'
  },
  container: {
    border: '2px solid #ababab',
    borderRadius: '4px',
    padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 4}px`,
    background: '#fafafa'
  },
  button: {
    marginTop: theme.spacing.unit * 3,
    cursor: 'pointer',
    background: '#eee',
    padding: theme.spacing.unit * 2,
    width: 'fit-content',
    borderRadius: '4px'
  }
});

export class CorrectResponse extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    model: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);

    const allTools = [
      { name: 'point', component: tools.point(), selected: true },
      { name: 'circle', component: tools.circle(), selected: true },
      { name: 'polygon', component: tools.polygon(), selected: true },
      { name: 'segment', component: tools.segment(), selected: true },
      { name: 'vector', component: tools.vector(), selected: true },
      { name: 'ray', component: tools.ray(), selected: true },
      { name: 'line', component: tools.line(), selected: true },
      { name: 'sine', component: tools.sine(), selected: true },
      { name: 'parabola', component: tools.parabola(), selected: true },
      { name: 'label', component: tools.label(), selected: true }
    ];

    const selectedTools = allTools.filter(t => t.selected);
    this.state = {
      allTools,
      currentTool: selectedTools && selectedTools[0],
      settings: {
        includeArrows: true,
        labels: true,
        graphTitle: false,
        coordinatesOnHover: false,
        size: {
          width: 600,
          height: 300
        }
      },
    };
  }

  changeMarks = (key, marks) => {
    const { model } = this.props;

    set(model, `answers.${key}.marks`, marks);
    this.props.onChange(model);
  };

  render() {
    const { classes, model } = this.props;
    const { allTools, currentTool, settings } = this.state;
    const selectedTools = allTools.filter(t => t.selected).map(t => t.component);
    console.log('model=', model);

    return (
      <div>
        Define Correct Response

        <div className={classes.graphingTools}>
          GRAPHING TOOLS

          <div className={classes.availableTools}>
            {allTools.map(tool => {
              return (
                <div
                  key={tool.name}
                  className={classnames(classes.availableTool, tool.selected && classes.selectedTool)}
                  onClick={() => {
                    const nextAllTools = allTools.map(t => {
                      if (tool.name === t.name) {
                        t.selected = !tool.selected
                      }

                      return t;
                    });

                    this.setState({
                      allTools: nextAllTools,
                      currentTool: (currentTool.name === tool.name && tool.selected) ? nextAllTools[0] : currentTool
                    });
                  }}
                >
                  {tool.name}
                </div>
              )
            })}
          </div>
        </div>

        <div className={classes.container}>
          {Object.keys(model.answers).map(mark => (
            <div key={`correct-response-graph-${model.answers[mark].name}`}>
              <p>{model.answers[mark].name}</p>

              <Graph
                size={settings.size}
                domain={model.domain}
                range={model.range}
                title={model.title}
                axesSettings={{
                  includeArrows: settings.includeArrows
                }}
                labels={settings.labels && model.labels}
                marks={model.answers[mark].marks}
                backgroundMarks={model.backgroundMarks}
                onChangeMarks={marks => this.changeMarks(mark, marks)}
                tools={selectedTools}
                currentTool={currentTool && currentTool.component}
                defaultTool={selectedTools && selectedTools[0] && selectedTools[0].type}
              />
            </div>
          ))}

          <div
            className={classes.button}
            onClick={() => {
              set(model, `answers.${`alternate${Object.keys(model.answers).length}`}`, {
                  name: `Alternate ${Object.keys(model.answers).length}`,
                  marks: []
                }
              );
              this.props.onChange(model);
            }}
          >
            ADD ALTERNATE
          </div>
        </div>
      </div>
    );
  }
}

export default withStyles(styles)(CorrectResponse);
