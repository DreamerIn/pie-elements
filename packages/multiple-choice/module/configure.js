import { _dll_react } from '@pie-ui/shared-lib@1.0.0/index.js';
// ...
