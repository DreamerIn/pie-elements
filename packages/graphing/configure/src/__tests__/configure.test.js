import * as React from 'react';
import { shallow } from 'enzyme';

import { Configure } from '../configure';
import { GraphingConfig } from '../graphing-config';
import { CorrectResponse } from '../correct-response';
import defaultValues from '../defaults';

jest.mock('@pie-lib/config-ui', () => ({
  InputContainer: props => <div>{props.children}</div>,
  InputCheckbox: props => <div>{props.children}</div>,
  FeedbackConfig: props => <div>{props.children}</div>,
  layout: {
    ConfigLayout: props => <div>{props.children}</div>
  },
  settings: {
    Panel: props => <div onChange={props.onChange} />,
    toggle: jest.fn(),
    radio: jest.fn(),
    numberFields: jest.fn()
  }
}));

jest.mock('@pie-lib/graphing', () => ({
  GraphContainer: props => <div>{props.children}</div>,
  tools: {
    point: () => <div />,
    circle: () => <div />,
    polygon: () => <div />,
    segment: () => <div />,
    vector: () => <div />,
    ray: () => <div />,
    line: () => <div />,
    sine: () => <div />,
    parabola: () => <div />,
    label: () => <div />,
  }
}));

describe('Configure', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = props => {
      const configureProps = { ...defaultValues, ...props };

      return shallow(<Configure { ...configureProps } />);
    };
  });

  describe('renders', () => {
    it('snapshot', () => {
      expect(wrapper()).toMatchSnapshot();
    })
  });

  describe('logic', () => {
    it('updates rationale', () => {
      const onModelChanged = jest.fn();
      const component = wrapper({ onModelChanged });

      component.instance().onRationaleChange('New Rationale');

      expect(onModelChanged).toBeCalledWith(expect.objectContaining({
        ...defaultValues.model,
        rationale: 'New Rationale',
      }));
    });

    it('updates prompt', () => {
      const onModelChanged = jest.fn();
      const component = wrapper({ onModelChanged });

      component.instance().onPromptChange('New Prompt');

      expect(onModelChanged).toBeCalledWith(expect.objectContaining({
        ...defaultValues.model,
        prompt: 'New Prompt',
      }));
    });
  });
});

describe('GraphingConfig', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      classes: {},
      model: defaultValues.model,
      onChange: jest.fn(),
    };

    wrapper = newProps => {
      const configureProps = { ...props, newProps };

      return shallow(<GraphingConfig { ...configureProps } />);
    };
  });

  describe('renders', () => {
    it('snapshot', () => {
      expect(wrapper()).toMatchSnapshot();
    })
  });

  describe('logic', () => {
    it('changeBackgroundMarks calls onChange', () => {
      const component = wrapper();
      const bM = [{ x: 1, y: 1, type: 'point'}];

      component.instance().changeBackgroundMarks(bM);

      expect(component.instance().props.onChange).toBeCalledWith({
        ...defaultValues.model,
        backgroundMarks: bM
      });
    });

    it('onChangeInputValue calls onChange', () => {
      const component = wrapper();
      component.instance().onChangeInputValue('domain.min', 20);

      expect(component.instance().props.model.domain.min).toEqual(20);
      expect(component.instance().props.onChange).toHaveBeenCalledWith({
        ...defaultValues.model,
        domain: {
          ...defaultValues.model.domain,
          min: 20
        }
      })
    });
  });
});


describe('CorrectResponse', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      classes: {},
      model: defaultValues.model,
      onChange: jest.fn(),
    };

    wrapper = newProps => {
      const configureProps = { ...props, newProps };

      return shallow(<CorrectResponse { ...configureProps } />);
    };
  });

  describe('renders', () => {
    it('snapshot', () => {
      expect(wrapper()).toMatchSnapshot();
    })
  });

  describe('logic', () => {
    it('changeMarks calls onChange', () => {
      const component = wrapper();
      const marks = [{ x: 1, y: 1, type: 'point'}];

      component.instance().changeMarks('alternateTest', marks);

      expect(component.instance().props.onChange).toBeCalledWith({
        ...defaultValues.model,
        answers: {
          ...defaultValues.model.answers,
          alternateTest: {
            marks
          }
        }
      });
    });

    it('changeDisplayedTools calls onChange', () => {
      const component = wrapper();
      component.instance().changeDisplayedTools([]);

      expect(component.instance().props.model.displayedTools).toEqual([]);
      expect(component.instance().props.onChange).toHaveBeenCalledWith({
        ...defaultValues.model,
        displayedTools: []
      })
    });
  });
});

