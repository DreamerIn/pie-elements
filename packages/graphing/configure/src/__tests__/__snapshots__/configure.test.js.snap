// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configure renders snapshot 1`] = `
<ConfigLayout
  settings={
    <Panel
      configuration={
        Object {
          "arrows": Object {
            "label": "Include arrows",
            "settings": false,
          },
          "authoring": Object {
            "enabled": false,
            "label": "Allow authoring",
            "settings": false,
          },
          "labels": Object {
            "enabled": true,
            "label": "Labels",
            "settings": false,
          },
          "padding": Object {
            "label": "Padding",
            "settings": false,
          },
          "prompt": Object {
            "label": "Item Stem",
            "settings": true,
          },
          "rationale": Object {
            "enabled": false,
            "label": "Rationale",
            "settings": true,
          },
          "scoringType": Object {
            "label": "Scoring Type",
            "settings": true,
          },
          "studentInstructions": Object {
            "enabled": true,
            "label": "Student Instructions",
            "settings": false,
          },
          "teacherInstructions": Object {
            "enabled": false,
            "label": "Teacher Instructions",
            "settings": false,
          },
          "title": Object {
            "enabled": true,
            "label": "Graph Title",
            "settings": false,
          },
        }
      }
      groups={
        Object {
          "Item Type": Object {
            "arrows": false,
            "graph": undefined,
            "labels": false,
            "padding": false,
            "title.enabled": false,
          },
          "Properties": Object {
            "authoring.enabled": false,
            "rationale.enabled": undefined,
            "scoringType": undefined,
            "studentInstructions.enabled": false,
            "teacherInstructions.enabled": false,
          },
        }
      }
      model={
        Object {
          "answers": Object {
            "a1": Object {
              "marks": Array [
                Object {
                  "type": "point",
                  "x": 1,
                  "y": 1,
                },
                Object {
                  "type": "point",
                  "x": 2,
                  "y": 2,
                },
                Object {
                  "from": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "to": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "type": "segment",
                },
              ],
              "name": "A1",
            },
            "a2": Object {
              "marks": Array [
                Object {
                  "type": "point",
                  "x": 1,
                  "y": 1,
                },
                Object {
                  "type": "point",
                  "x": 2,
                  "y": 2,
                },
                Object {
                  "type": "point",
                  "x": 3,
                  "y": 3,
                },
                Object {
                  "from": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "to": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "type": "segment",
                },
              ],
              "name": "A2",
            },
          },
          "arrows": true,
          "backgroundMarks": Array [],
          "domain": Object {
            "axisLabel": "x",
            "labelStep": 1,
            "max": 10,
            "min": -10,
            "padding": 0,
            "step": 1,
          },
          "graph": Object {
            "height": 480,
            "width": 480,
          },
          "labels": null,
          "padding": true,
          "prompt": "Here goes item stem!",
          "range": Object {
            "axisLabel": "y",
            "labelStep": 1,
            "max": 5,
            "min": -5,
            "padding": 0,
            "step": 1,
          },
          "rationale": "Rationale goes here!",
          "scoringType": "partial scoring",
          "title": "",
        }
      }
    />
  }
>
  <div>
    <WithStyles(Typography)
      component="div"
      type="body1"
    >
      <span>
        This interaction asks a student to draw a line that meets specific criteria. The student will draw the line by clicking on two points on the graph.
      </span>
    </WithStyles(Typography)>
    <InputContainer
      label="Item Stem"
    >
      <EditableHtml
        disableUnderline={true}
        markup="Here goes item stem!"
        nonEmpty={false}
        onChange={[Function]}
      />
    </InputContainer>
    <WithStyles(GraphingConfig)
      authoringEnabled={false}
      config={
        Object {
          "height": 480,
          "width": 480,
        }
      }
      model={
        Object {
          "answers": Object {
            "a1": Object {
              "marks": Array [
                Object {
                  "type": "point",
                  "x": 1,
                  "y": 1,
                },
                Object {
                  "type": "point",
                  "x": 2,
                  "y": 2,
                },
                Object {
                  "from": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "to": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "type": "segment",
                },
              ],
              "name": "A1",
            },
            "a2": Object {
              "marks": Array [
                Object {
                  "type": "point",
                  "x": 1,
                  "y": 1,
                },
                Object {
                  "type": "point",
                  "x": 2,
                  "y": 2,
                },
                Object {
                  "type": "point",
                  "x": 3,
                  "y": 3,
                },
                Object {
                  "from": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "to": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "type": "segment",
                },
              ],
              "name": "A2",
            },
          },
          "arrows": true,
          "backgroundMarks": Array [],
          "domain": Object {
            "axisLabel": "x",
            "labelStep": 1,
            "max": 10,
            "min": -10,
            "padding": 0,
            "step": 1,
          },
          "graph": Object {
            "height": 480,
            "width": 480,
          },
          "labels": null,
          "padding": true,
          "prompt": "Here goes item stem!",
          "range": Object {
            "axisLabel": "y",
            "labelStep": 1,
            "max": 5,
            "min": -5,
            "padding": 0,
            "step": 1,
          },
          "rationale": "Rationale goes here!",
          "scoringType": "partial scoring",
          "title": "",
        }
      }
    />
    <WithStyles(CorrectResponse)
      config={
        Object {
          "height": 480,
          "width": 480,
        }
      }
      model={
        Object {
          "answers": Object {
            "a1": Object {
              "marks": Array [
                Object {
                  "type": "point",
                  "x": 1,
                  "y": 1,
                },
                Object {
                  "type": "point",
                  "x": 2,
                  "y": 2,
                },
                Object {
                  "from": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "to": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "type": "segment",
                },
              ],
              "name": "A1",
            },
            "a2": Object {
              "marks": Array [
                Object {
                  "type": "point",
                  "x": 1,
                  "y": 1,
                },
                Object {
                  "type": "point",
                  "x": 2,
                  "y": 2,
                },
                Object {
                  "type": "point",
                  "x": 3,
                  "y": 3,
                },
                Object {
                  "from": Object {
                    "x": 1,
                    "y": 1,
                  },
                  "to": Object {
                    "x": 2,
                    "y": 2,
                  },
                  "type": "segment",
                },
              ],
              "name": "A2",
            },
          },
          "arrows": true,
          "backgroundMarks": Array [],
          "domain": Object {
            "axisLabel": "x",
            "labelStep": 1,
            "max": 10,
            "min": -10,
            "padding": 0,
            "step": 1,
          },
          "graph": Object {
            "height": 480,
            "width": 480,
          },
          "labels": null,
          "padding": true,
          "prompt": "Here goes item stem!",
          "range": Object {
            "axisLabel": "y",
            "labelStep": 1,
            "max": 5,
            "min": -5,
            "padding": 0,
            "step": 1,
          },
          "rationale": "Rationale goes here!",
          "scoringType": "partial scoring",
          "title": "",
        }
      }
    />
  </div>
</ConfigLayout>
`;

exports[`CorrectResponse renders snapshot 1`] = `
<div>
  Define Correct Response
  <div>
    GRAPHING TOOLS
    <div>
      <div
        className=""
        key="point"
        onClick={[Function]}
      >
        point
      </div>
      <div
        className=""
        key="circle"
        onClick={[Function]}
      >
        circle
      </div>
      <div
        className=""
        key="polygon"
        onClick={[Function]}
      >
        polygon
      </div>
      <div
        className=""
        key="segment"
        onClick={[Function]}
      >
        segment
      </div>
      <div
        className=""
        key="vector"
        onClick={[Function]}
      >
        vector
      </div>
      <div
        className=""
        key="ray"
        onClick={[Function]}
      >
        ray
      </div>
      <div
        className=""
        key="line"
        onClick={[Function]}
      >
        line
      </div>
      <div
        className=""
        key="sine"
        onClick={[Function]}
      >
        sine
      </div>
      <div
        className=""
        key="parabola"
        onClick={[Function]}
      >
        parabola
      </div>
    </div>
  </div>
  <div>
    <div
      key="correct-response-graph-A1"
    >
      <p>
        A1
      </p>
      <GraphContainer
        backgroundMarks={Array []}
        currentTool={<div />}
        defaultTool="div"
        displayedTools={
          Array [
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
          ]
        }
        domain={
          Object {
            "axisLabel": "x",
            "labelStep": 1,
            "max": 10,
            "min": 20,
            "padding": 0,
            "step": 1,
          }
        }
        labels={null}
        marks={
          Array [
            Object {
              "type": "point",
              "x": 1,
              "y": 1,
            },
            Object {
              "type": "point",
              "x": 2,
              "y": 2,
            },
            Object {
              "from": Object {
                "x": 1,
                "y": 1,
              },
              "to": Object {
                "x": 2,
                "y": 2,
              },
              "type": "segment",
            },
          ]
        }
        onChangeMarks={[Function]}
        range={
          Object {
            "axisLabel": "y",
            "labelStep": 1,
            "max": 5,
            "min": -5,
            "padding": 0,
            "step": 1,
          }
        }
        size={
          Object {
            "height": 480,
            "width": 480,
          }
        }
        title=""
        tools={
          Array [
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
          ]
        }
      />
    </div>
    <div
      key="correct-response-graph-A2"
    >
      <p>
        A2
      </p>
      <GraphContainer
        backgroundMarks={Array []}
        currentTool={<div />}
        defaultTool="div"
        displayedTools={
          Array [
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
          ]
        }
        domain={
          Object {
            "axisLabel": "x",
            "labelStep": 1,
            "max": 10,
            "min": 20,
            "padding": 0,
            "step": 1,
          }
        }
        labels={null}
        marks={
          Array [
            Object {
              "type": "point",
              "x": 1,
              "y": 1,
            },
            Object {
              "type": "point",
              "x": 2,
              "y": 2,
            },
            Object {
              "type": "point",
              "x": 3,
              "y": 3,
            },
            Object {
              "from": Object {
                "x": 1,
                "y": 1,
              },
              "to": Object {
                "x": 2,
                "y": 2,
              },
              "type": "segment",
            },
          ]
        }
        onChangeMarks={[Function]}
        range={
          Object {
            "axisLabel": "y",
            "labelStep": 1,
            "max": 5,
            "min": -5,
            "padding": 0,
            "step": 1,
          }
        }
        size={
          Object {
            "height": 480,
            "width": 480,
          }
        }
        title=""
        tools={
          Array [
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
          ]
        }
      />
    </div>
    <div
      onClick={[Function]}
    >
      ADD ALTERNATE
    </div>
  </div>
</div>
`;

exports[`GraphingConfig renders snapshot 1`] = `
<div>
  Define Graph Attributes
  <div>
    <div
      key="graph"
    >
      <WithStyles(Typography)
        component="div"
        type="body1"
      >
        <span>
          Use the tools below to set background shapes
        </span>
      </WithStyles(Typography)>
      <GraphContainer
        backgroundMarks={Array []}
        currentTool={<div />}
        defaultTool="div"
        domain={
          Object {
            "axisLabel": "x",
            "labelStep": 1,
            "max": 10,
            "min": -10,
            "padding": 0,
            "step": 1,
          }
        }
        key="graphing-config"
        labels={null}
        marks={Array []}
        onChangeMarks={[Function]}
        range={
          Object {
            "axisLabel": "y",
            "labelStep": 1,
            "max": 5,
            "min": -5,
            "padding": 0,
            "step": 1,
          }
        }
        size={
          Object {
            "height": 480,
            "width": 480,
          }
        }
        title=""
        tools={
          Array [
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
            <div />,
          ]
        }
      />
    </div>
  </div>
</div>
`;
